pgjdbc/src/main/java/org/postgresql/util/PasswordUtil.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import org.postgresql.core.Utils;
+
+import com.ongres.scram.common.ScramFunctions;
+import com.ongres.scram.common.ScramMechanisms;
+import com.ongres.scram.common.bouncycastle.base64.Base64;
+import com.ongres.scram.common.stringprep.StringPreparations;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class PasswordUtil {
+  private static final String DEFAULT_PASSWORD_ENCRYPTION = "scram-sha-256";
+  private static final int DEFAULT_ITERATIONS = 4096;
+  private static final int DEFAULT_SALT_LENGTH = 16;
+
+  private static class SecureRandomHolder {
+    static final SecureRandom INSTANCE = new SecureRandom();
+  }
+
+  private static SecureRandom getSecureRandom() {
+    return SecureRandomHolder.INSTANCE;
+  }
+
+  /**
+   * Generate the encoded text representation of the given password for
+   * SCRAM-SHA-256 authentication. The return value of this method is the literal
+   * text that may be used when creating or modifying a user with the given
+   * password without the surrounding single quotes.
+   *
+   * @param password   The plain text of the user's password
+   * @param iterations The number of iterations of the hashing algorithm to
+   *                   perform
+   * @param salt       The random salt value
+   * @return The text representation of the password encrypted for SCRAM-SHA-256
+   *         authentication
+   */
+  public static String encodeScramSha256(char[] password, int iterations, byte[] salt) {
+    Objects.requireNonNull(password, "password");
+    if (iterations <= 0) {
+      throw new IllegalArgumentException("iterations must be greater than zero");
+    }
+    if (salt.length == 0) {
+      throw new IllegalArgumentException("salt length must be greater than zero");
+    }
+    try {
+      String passwordText = String.valueOf(password);
+      byte[] saltedPassword = ScramFunctions.saltedPassword(ScramMechanisms.SCRAM_SHA_256,
+          StringPreparations.SASL_PREPARATION, passwordText, salt, iterations);
+      byte[] clientKey = ScramFunctions.clientKey(ScramMechanisms.SCRAM_SHA_256, saltedPassword);
+      byte[] storedKey = ScramFunctions.storedKey(ScramMechanisms.SCRAM_SHA_256, clientKey);
+      byte[] serverKey = ScramFunctions.serverKey(ScramMechanisms.SCRAM_SHA_256, saltedPassword);
+
+      return "SCRAM-SHA-256" //
```

Comment:
Why adding empty comments?

Reply:
Why show up with feedback on lines that have not changed since the PR was first opened after the PR is merged?

Are you just looking for something to nitpick?

Reply:
You did not fix javadocs (see https://github.com/pgjdbc/pgjdbc/pull/3082#discussion_r1437354429), and I thought it would be easier and faster to just wait for you to merge the PR and then fix the style issues